### PR TITLE
fluxcd: add updateScript

### PR DESCRIPTION
**Pull request by superherointj**
* fluxcd: add updateScript

### CODE REVIEW

File: pkgs/applications/networking/cluster/fluxcd/default.nix [hunk 1]
```diff
@@ -50,6 +51,8 @@ buildGoModule rec {
     done
   '';
 
+  passthru.updateScript = ./update.sh;
```
Comment (r-burns) [inline pkgs/applications/networking/cluster/fluxcd/default.nix:54]: Does nix-update-script not work here?

Comment (superherointj) [reply pkgs/applications/networking/cluster/fluxcd/default.nix:54]: Doesn't handle the extra manifest. Or does it?

Comment (r-burns) [reply pkgs/applications/networking/cluster/fluxcd/default.nix:54]: Ah right, I would guess not.

File: pkgs/applications/networking/cluster/fluxcd/default.nix [hunk 2]
```diff
@@ -1,25 +1,26 @@
 { lib, buildGoModule, fetchFromGitHub, fetchzip, installShellFiles }:
 
 let
-  version = "0.17.2";
+  pkgVersion = "0.17.2";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/fluxcd/default.nix:4]:
```suggestion
  version = "0.17.2";
```
Renaming the others should be fine I guess.

File: pkgs/applications/networking/cluster/fluxcd/update.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl jq
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/fluxcd/update.sh:2]:
```suggestion
#!nix-shell -i bash -p curl gnused jq
```

File: pkgs/applications/networking/cluster/fluxcd/update.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+setKV pkgVersion ${LATEST_VERSION}
+setKV pkgSha256 ${PKG_SHA256}
+setKV manifestsSha256 ${SPEC_SHA256}
+
+cd ../../../../../
+set +e
+VENDOR_SHA256=$(nix-build --no-out-link -A fluxcd 2>&1 | grep "got:" | cut -d':' -f2 | sed 's/ //g')
+set -e
+
+cd pkgs/applications/networking/cluster/fluxcd
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/fluxcd/update.sh:31]:
```suggestion
cd -
```

Comment (superherointj) [reply pkgs/applications/networking/cluster/fluxcd/update.sh]: Done.

File: pkgs/applications/networking/cluster/fluxcd/update.sh (hunk 3 above)
Comment (superherointj) [reply pkgs/applications/networking/cluster/fluxcd/update.sh]: Done

File: pkgs/applications/networking/cluster/fluxcd/default.nix (hunk 2 above)
Comment (superherointj) [reply pkgs/applications/networking/cluster/fluxcd/default.nix]: Done

File: pkgs/applications/networking/cluster/fluxcd/default.nix [hunk 5]
```diff
@@ -2,24 +2,25 @@
...
-
   pname = "fluxcd";
+  inherit version;
 
   src = fetchFromGitHub {
     owner = "fluxcd";
     repo = "flux2";
     rev = "v${version}";
-    sha256 = "0kcdx4ldnshk4pqq37a7p08xr5cpsjrbrifk9fc3jbiw39m09mhf";
+    sha256 = pkgSha256;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/fluxcd/default.nix:23]:
```suggestion
    sha256 = sha256;
```

File: pkgs/applications/networking/cluster/fluxcd/default.nix [hunk 6]
```diff
@@ -2,24 +2,25 @@
 
 let
   version = "0.17.2";
+  pkgSha256 = "0kcdx4ldnshk4pqq37a7p08xr5cpsjrbrifk9fc3jbiw39m09mhf";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/fluxcd/default.nix:5]:
```suggestion
  sha256 = "0kcdx4ldnshk4pqq37a7p08xr5cpsjrbrifk9fc3jbiw39m09mhf";
```

File: pkgs/applications/networking/cluster/fluxcd/update.sh [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+PKG_SHA256=$(nix-prefetch-url --quiet --unpack https://github.com/fluxcd/flux2/archive/refs/tags/${LATEST_TAG}.tar.gz)
+
+SPEC_SHA256=$(nix-prefetch-url --quiet --unpack https://github.com/fluxcd/flux2/releases/download/${LATEST_TAG}/manifests.tar.gz)
+
+setKV () {
+  sed -i "s/$1 = \".*\"/$1 = \"$2\"/" ./default.nix
+}
+
+setKV version ${LATEST_VERSION}
+setKV pkgSha256 ${PKG_SHA256}
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/fluxcd/update.sh:21]:
```suggestion
setKV sha256 ${PKG_SHA256}
```

File: pkgs/applications/networking/cluster/fluxcd/default.nix [hunk 8]
```diff
@@ -2,24 +2,25 @@
...
-
   pname = "fluxcd";
+  inherit version;
 
   src = fetchFromGitHub {
     owner = "fluxcd";
     repo = "flux2";
     rev = "v${version}";
-    sha256 = "0kcdx4ldnshk4pqq37a7p08xr5cpsjrbrifk9fc3jbiw39m09mhf";
+    sha256 = sha256;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/fluxcd/default.nix:23]:
```suggestion
    inherit sha256;
```

**Pull request merged**